src/components/MultiSelect/MultiSelectListbox.tsx

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import {
+  Box,
+  ListItem,
+  UnorderedList,
+  useMultiStyleConfig,
+} from "@chakra-ui/react";
+import Checkbox from "./../Checkbox/Checkbox";
+import { MultiSelectItem } from "./MultiSelectTypes";
+import { MultiSelectProps } from "./MultiSelect";
+import MultiSelectMenuButton from "./MultiSelectMenuButton";
+import { useSelect } from "downshift";
+
+type MultiSelectListboxProps = Omit<MultiSelectProps, "onChange"> & {
+  onChange: (selectedItem: MultiSelectItem, id: string) => void;
+};
+
+function MultiSelectListbox({
+  id,
+  label,
+  defaultIsOpen,
+  items,
+  onChange,
+  selectedItems,
+  onClear,
+  width = "default",
+  isBlockElement = false,
+}: MultiSelectListboxProps) {
+  const styles = useMultiStyleConfig("MultiSelect", {
+    width,
+    isBlockElement,
+  });
+  // Downshift.
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getMenuProps,
+    highlightedIndex,
+    getItemProps,
+  } = useSelect({
+    items,
+    // Downshift's internal state for handling keyboard and mouse events.
+    stateReducer: (state, actionAndChanges) => {
+      const { changes, type } = actionAndChanges;
+      switch (type) {
+        case useSelect.stateChangeTypes.MenuKeyDownEnter:
+        case useSelect.stateChangeTypes.MenuKeyDownSpaceButton:
+        case useSelect.stateChangeTypes.ItemClick:
+          return {
+            ...changes,
+            isOpen: true, // Keep menu open after selection.
+            highlightedIndex: state.highlightedIndex,
+          };
+        default:
+          return changes;
+      }
+    },
+    // @ts-ignore
+    selectedItem: selectedItems,
+    onSelectedItemChange: ({ selectedItem }) => {
+      onChange(selectedItem, id);
+    },
+    initialIsOpen: defaultIsOpen,
+  });
+
+  return (
+    <Box id={id} __css={styles}>
+      <MultiSelectMenuButton
+        multiSelectId={id}
+        label={label}
+        isOpen={isOpen}
+        selectedItems={selectedItems}
+        onClear={onClear}
+        {...getToggleButtonProps({
+          // @TODO add short prefix to avoid id starting with number
+          id: `${id}-toggle-button`,
+          "aria-labelledby": `${id}-label ${id}-toggle-button`,
+        })}
+      />
+      <Box __css={styles.menuContainer} {...(!isOpen && { display: "none" })}>
```

Comment:
Instead of doing the logic check here for `{...(!isOpen && { display: "none" })}`, pass `isOpen` to the `useMultiStyleConfig` function and update the styles in the `styles.menuContainer` object.